admin-api-server/src/nft/nft.controller.ts

```diff
@@ -23,70 +23,74 @@ import { NftService } from './nft.service';
 import { CurrentUser } from '../decoraters/user.decorator';
 import { User } from 'src/user/entities/user.entity';
 import { NftPaginationParams, NftFilterParams } from './params';
+import { ParseJSONArrayPipe } from 'src/pipes/ParseJSONArrayPipe';
+import { FilterParams } from 'src/types';
 
-interface UpdateNft {
+export interface UpdateNft {
   attribute: string;
   value?: string;
 }
 
 @Controller('nft')
 export class NftController {
-  constructor(private readonly nftService: NftService) {}
+  constructor(private readonly nftService: NftService) { }
 
   @Post()
   @UseGuards(JwtAuthGuard)
-  create(@CurrentUser() user: User) {
-    return this.nftService.create(user);
+  async create(@CurrentUser() user: User) {
+    return await this.nftService.create(user);
   }
 
   @Get()
-  async findAll(@Query() params: NftFilterParams) {
-    this.#validatePaginationParams(params);
-    return this.nftService.findAll(params);
+  @UseGuards(JwtAuthGuard)
+  async findAll(
+    @Query('sort', new ParseJSONArrayPipe())
+    sort?: string[],
+    @Query('filter', new ParseJSONArrayPipe()) filter?: FilterParams,
+    @Query('range', new ParseJSONArrayPipe())
+    range?: number[],
+  ) {
+    // this.#validatePaginationParams(params);
+    return this.nftService.findAll({
+      sort,
+      filter,
+      range
+    });
   }
 
-  @UseGuards(JwtAuthGuard)
   @Get(':id')
+  @UseGuards(JwtAuthGuard)
   async findOne(@Param('id') id: number, @CurrentUser() user: User) {
     return await this.nftService.getNft(user, id);
   }
 
   @UseInterceptors(
-    FileInterceptor('data', {
+    FileInterceptor('picture', {
       limits: { fileSize: NFT_IMAGE_MAX_BYTES },
     }),
   )
-  @Patch(':id/setContent')
-  @UseGuards(JwtAuthGuard)
-  async setContent(
-    @Param('id') nftId: number,
-    @CurrentUser() user: User,
-    @UploadedFile() picture: any,
-  ): Promise<NftEntity> {
-    if (typeof picture === 'undefined') {
-      throw new HttpException(`expected file upload`, HttpStatus.BAD_REQUEST);
-    }
-    return await this.nftService.setContent(user, nftId, picture);
-  }
-
   @UseGuards(JwtAuthGuard)
   @Patch(':id')
   async update(
     @Param('id') nftId: number,
     @Body() updateNft: UpdateNft,
     @CurrentUser() user: User,
+    @UploadedFile() picture: any
   ): Promise<NftEntity> {
+
     if (updateNft.attribute === this.nftService.CONTENT_KEYWORD) {
-      throw new HttpException(
-        `attribute=${this.nftService.CONTENT_KEYWORD} is a reserved attribute that is not allowed to be set through this endpoint`,
-        HttpStatus.BAD_REQUEST,
-      );
+      await this.nftService.setContent(user, nftId, updateNft.value);
```

Comment:
If there is a picture to be uploaded, the other remaining (if any) updateNft attributes+values should be applied inside the same "stm transaction" (in quotes because it's not called that way in the code but it basically is a transactional application of stm, locked by nft ids). Therefore `updateNft` should be passed down to `setContent` and `setContent` should apply it within it's nft id lock. And then here it should return after the `setContent` call.